ragna/source_storages/_embedding_model.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+from sentence_transformers import SentenceTransformer
+import torch
+
+from ragna.core._components import Component
+
+from ragna.source_storages._vector_database import Chunk
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+
+class Embedding:
+    embedding: list[float]
+    chunk: Chunk
+
+    def __init__(self, embedding: list[float], chunk: Chunk):
+        super().__init__()
+        self.embedding = embedding
+        self.chunk = chunk
+
+
+class GenericEmbeddingModel(Component, ABC):
+    _EMBEDDING_DIMENSIONS: int
+
+    @abstractmethod
+    def embed_chunks(self, chunks: list[Chunk]) -> list[Embedding]:
+        raise NotImplementedError
+
+    def embed_text(self, text: str) -> list[float]:
+        raise NotImplementedError
```

Comment:
Is there ever a use case to do that? Aren't we always going to embed chunks?

Reply:
No, because when you prompt the database you must convert your prompt into an embedding when you query.  Either we turn this prompt into a chunk which I don't like, or just separate the logic like this.